### PR TITLE
New travis php symfony

### PR DESCRIPTION
**Pull request by phil-davis**
Make the Travis matrix a bit more "modern" (not that PHP 5.6 is all that modern).

Eliminates the PHP 5.3. Travis job that fails.

### CODE REVIEW

**Comment by phil-davis** [comment]
Note: HHVM will continue to support PHP5.6 for its security-patching life (end of 2018)
http://hhvm.com/blog/2017/09/18/the-future-of-hhvm.html
http://php.net/supported-versions.php

So the HHVM job needs to be removed when PHP5.6 jobs are removed from the Travis matrix, or it could be removed now.

**Comment by phil-davis** [comment]
Symfony version support is here:
http://symfony.com/doc/current/contributing/community/releases.html
under heading "Schedule".

**Comment by phil-davis** [comment]
@stof comments?

File: .travis.yml [hunk 1]
```diff
@@ -7,16 +7,15 @@ cache:
     - $HOME/.composer/cache/files
 
 php:
-  - 5.3
-  - 5.4
```
Comment (stof) [inline .travis.yml]: Removing old PHP version is a no-go if we don't change the PHP requirement in the composer.json. Not having CI for these versions means we would not be able to ensure comaptibility, so it is better to tell composer to use an older release for these versions (Behat itself made this exact mistake in 3.4).

File: .travis.yml [hunk 2]
```diff
@@ -7,16 +7,15 @@ cache:
...
+  - 7.1
 
 matrix:
   include:
-    - php: 5.5
-      env: SYMFONY_VERSION='2.3.*'
+    - php: 5.6
+      env: SYMFONY_VERSION='2.8.*'
     # Test against dev dependencies
     - php: 5.6
```
Comment (stof) [inline .travis.yml]: I suggest moving this job to 7.1 (latest dev version of Symfony is not compatible with 5.6 for instance)

File: .travis.yml [hunk 3]
```diff
@@ -7,16 +7,15 @@ cache:
     - $HOME/.composer/cache/files
 
 php:
-  - 5.3
-  - 5.4
-  - 5.5
   - 5.6
   - hhvm
```
Comment (stof) [inline .travis.yml]: I suggest removing the HHVM job, given that HHVM does not care about PHP compatibility anymore: http://hhvm.com/blog/2017/09/18/the-future-of-hhvm.html

File: .travis.yml (hunk 1 above)
Comment (phil-davis) [reply .travis.yml]: something like #297 ?

**Comment by phil-davis** [comment]
Started getting a bunch of little commits here. I cloned the repo rather than continuing to edit online, and made PR #297 with a single commit. Tat contains all the changes suggested as I understand them.

I am not sure if you want to keep official support for PHP 5.3 5.4 5.5 - if so, then of course those Travis matrix jobs need to be kept and the now-failing job for PHP 5.3 would have to be fixed up so it passes.

**Pull request closed without merging**